### PR TITLE
Render preview placement as a tab instead of an overlay

### LINKED ISSUE #487: Bug: preview placement renders over tabs instead of inside a tab
## Bug

The preview placement currently renders over the top of all tab content instead of inside its own tab. This makes all existing tabs (Shell, agent sessions, etc.) unusable while the preview is visible.

## Expected behaviour

Preview should render inside a tab in the tab bar alongside other tabs, just like Shell/agent tabs. Switching to another tab should hide the preview; switching back should show it.

## Actual behaviour

Preview renders as an overlay covering the entire terminal area. Tabs behind it are inaccessible.

## Screenshot

preview-overlay-bug

### PR DESCRIPTION
**Pull request by tomcorke**
## Summary
- The `preview` detail-view placement (introduced in #484) rendered as an absolute-positioned overlay with `z-index: 5`, covering all terminal tab content and making Shell and agent sessions unreachable while the preview was showing.
- This change reworks the preview to mount into a sibling host slot owned by `TerminalPanelView`, toggled by a new "Preview" pseudo-tab in the tab bar - mirroring the embedded placement from #483.
- Selecting the Preview tab shows the preview and hides the terminal wrapper; selecting any other tab flips back. Switching the placement setting away from `preview` restores the terminal wrapper automatically. Auto-close on item change is honoured the same way as the embedded placement.

## Implementation notes
- `TerminalPanelView` gains `getPreviewDetailHost` / `activatePreviewDetail` / `deactivatePreviewDetail` / `isPreviewDetailActive`, plus a Preview pseudo-tab rendered in `renderTabBar` when placement is `preview` and an item is active.
- `AdapterBundle.createDetailView` grows an optional `previewHost` parameter (alongside the existing `embeddedHost`). `TaskAgentAdapter` forwards it to `TaskDetailView.show`, which prefers the framework host and falls back to the legacy `.wt-terminal-wrapper` query for older callers.
- `TaskDetailView` now also applies `autoClose` to the preview (detach on item change) using the same `lastItemId` pattern as the leaf and embedded paths.
- `TaskPreviewView` itself is unchanged aside from doc comments - it was already host-agnostic.
- CSS drops `position: absolute` / `z-index: 5` from `.wt-preview-overlay` and adds `.wt-preview-detail-host` so the preview is a plain block inside its sibling slot, not layered above the terminal.

## Test plan
- [x] `pnpm run build` passes.
- [x] `pnpm exec vitest run` passes (1290 tests, 11 new).
- [x] New tests cover: Preview pseudo-tab only renders when placement is `preview` and an item is active; clicking it hides the terminal wrapper; clicking a terminal tab flips back; switching placement away from preview restores terminal visibility; autoClose detaches/remounts on item change; framework-supplied host is passed through to `TaskPreviewView`.
- [ ] Manual: switch placement to Preview, confirm Preview pseudo-tab appears, click Shell tab while preview is active, verify shell becomes visible again.

Closes #487

### CODE REVIEW

File: src/framework/TerminalPanelView.ts [hunk 1]
```diff
@@ -147,9 +155,22 @@ export class TerminalPanelView {
...
     if (this.embeddedDetailActive && !this.isEmbeddedPlacementActive()) {
       this.deactivateEmbeddedDetail();
-    } else {
+      tabBarRerendered = true;
+    }
+    if (this.previewDetailActive && !this.isPreviewPlacementActive()) {
+      this.deactivatePreviewDetail();
+      tabBarRerendered = true;
+    }
+    if (!tabBarRerendered) {
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply src/framework/TerminalPanelView.ts:173]: Fixed in 4630748 - MainView now detaches the adapter's detail view and re-runs createDetailView for the current selection whenever core.detailViewPlacement changes, so TaskPreviewView's vault modify listener is released immediately on switch-away and the new placement is mounted without requiring a reselect.

File: src/adapters/task-agent/TaskDetailView.ts [hunk 2]
```diff
@@ -65,10 +66,20 @@ export class TaskDetailView {
...
+        this.previewView.detach();
+        this.previewView = null;
+      }
       this.lastItemId = item.id;
-      const hostEl = this.resolvePreviewHost(ownerLeaf);
+      // Prefer the framework-supplied host (the sibling preview slot owned
+      // by TerminalPanelView). Fall back to the legacy wrapper query when
+      // the framework does not supply one - keeps older callers working.
+      const hostEl = previewHost ?? this.resolvePreviewHost(ownerLeaf);
       if (!hostEl) return;
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply src/adapters/task-agent/TaskDetailView.ts]: Fixed in d5c9cb9 - deleted the resolvePreviewHost(ownerLeaf) fallback entirely. The framework now always supplies previewHost via MainView, so the fallback path was dead code; keeping it would have silently mounted into the non-flex .wt-terminal-wrapper and broken the new flex-based layout. TaskDetailView.show() now warns and no-ops if previewHost is missing, flagging any future programming error loudly rather than rendering into the wrong container.
